app/src/main/java/com/codesoom/assignment/models/TasksStorage.java

```diff
@@ -0,0 +1,47 @@
+package com.codesoom.assignment.models;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Optional;
+
+/**
+ * 할 일 목록
+ */
+public class TasksStorage {
+    private final HashMap<Long, Task> tasks = new HashMap<>();
+    private Long newId;
+
+    public TasksStorage() {
+        newId = 0L;
+    }
+
+    public synchronized Task create(String title) {
+        Task task = new Task(newId, title);
+        tasks.put(task.getId(), task);
+
+        newId++;
+
+        return task;
+    }
+
+    public synchronized Optional<Task> read(Long id) {
+        return Optional.ofNullable(tasks.get(id));
+    }
+
+    public synchronized Collection<Task> readAll() {
+        return tasks.values();
+    }
+
+    public synchronized Optional<Task> update(Long id, String title) {
+        if (!tasks.containsKey(id)) {
+            return Optional.empty();
+        }
+        tasks.replace(id, new Task(id, title));
+
+        return Optional.of(new Task(id, title));
+    }
```

Comment:
`Optional`을 사용하지 않아도 괜찮을 것 같아요.

```suggestion
    public synchronized Optional<Task> update(Long id, String title) {
        if (!tasks.containsKey(id)) {
            throw new NotFoundException();
        }
        
        Task task = new Task(id, title);
        tasks.replace(id, task);
        return task;
    }
```